index.js

```diff
@@ -1,78 +1,106 @@
+module.exports = SignalHub
 
-module.exports = function (app, urls) {
+function SignalHub (app, urls) {
+  if (!(this instanceof SignalHub)) return new SignalHub(app, urls)
+
+  events.EventEmitter.call(this)
+  this.setMaxListeners(0)
+  
   if (!app) throw new Error('app name required')
   if (!urls || !urls.length) throw new Error('signalhub url(s) required')
 
-  var that = {}
   if (!Array.isArray(urls)) urls = [urls]
 
-  urls = urls.map(function (url) {
+  this.app = app
+  this.urls = urls.map(function (url) {
     url = url.replace(/\/$/, '')
     return url.indexOf('://') === -1 ? 'http://' + url : url
   })
+  this.subscribers = []
+}
 
-  that.subscribe = function (channel) {
-    var endpoint
-
-    if (Array.isArray(channel)) {
-      endpoint = channel.join(',')
-    } else {
-      endpoint = channel
-    }
-
-    var all = through.obj()
-    var streams = urls.map(function (url) {
-      return ess(url + '/v1/' + app + '/' + endpoint, {json: true})
-    })
+inherits(SignalHub, events.EventEmitter)
 
-    if (streams.length === 1) return streams[0]
+SignalHub.prototype.subscribe = function (channel) {
+  var endpoint = Array.isArray(channel) ? channel.join(',') : channel
 
-    var all = through.obj()
+  var streams = this.urls.map(function (url) {
+    return ess(url + '/v1/' + this.app + '/' + endpoint, {json: true})
+  })
 
-    all.setMaxListeners(0)
+  var subscriber
+  if (streams.length === 1) {
+    subscriber = streams[0]
+  } else {
+    subscriber = through.obj()
+    subscriber.setMaxListeners(0)
     streams.forEach(function (stream) {
       stream.on('open', function () {
-        all.emit('open')
+        subscriber.emit('open')
       })
-      pump(stream, all)
+      pump(stream, subscriber)
     })
-
-    return all
   }
 
-  var broadcast = function (url, channel, message, cb) {
-    nets({
-      method: 'POST',
-      json: message,
-      url: url + '/v1/' + app + '/' + channel
-    }, function (err, res) {
-      if (err) return cb(err)
-      if (res.statusCode !== 200) return cb(new Error('Bad status: ' + res.statusCode))
-      cb()
-    })
-  }
+  this.subscribers.push(subscriber)
 
-  that.broadcast = function (channel, message, cb) {
-    if (!cb) cb = noop
+  return subscriber
+}
+
+SignalHub.prototype.broadcast = function (channel, message, cb) {
+  if (!cb) cb = noop
+
+  var pending = this.urls.length
+  var errors = 0
 
-    var pending = urls.length
-    var errors = 0
+  var self = this
+  this.urls.forEach(function (url) {
+    broadcast(url, channel, message, function (err) {
+      if (err) errors++
+      if (--pending) return
+      if (errors === self.urls.length) return cb(err)
+      cb()
+    })
+  })
+}
 
-    urls.forEach(function (url) {
-      broadcast(url, channel, message, function (err) {
-        if (err) errors++
-        if (--pending) return
-        if (errors === urls.length) return cb(err)
-        cb()
+SignalHub.prototype.close = function (cb) {
```

Comment:
you can add `if (cb) this.once('close', cb)` and then get rid of the cb calls below